### PR TITLE
Fixes #37974 - support keep-latest-packages in python

### PR DESCRIPTION
**Pull request by sbernhard**
The pulp python backend supports keep-latest-packages. This should be available on the UI, too.

Thanks @lumarel  for adding the number content type some days ago. Very appreciated!

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
Not a complete review, just something that jumped out to me.

File: lib/katello/repository_types/python.rb [hunk 1]
```diff
@@ -28,6 +28,9 @@
   generic_remote_option :package_types, title: N_("Package Types"), type: Array, input_type: "text", delimiter: ",", default: [],
                         description: N_("Package types to sync for Python content, separated by comma. Leave empty to get every package type. Package types are: bdist_dmg, bdist_dumb, bdist_egg, bdist_msi, bdist_rpm, bdist_wheel, bdist_wininst, sdist.")
 
+  generic_remote_option :keep_latest_packages, title: N_("Keep latest packages"), type: Integer, input_type: "number", delimiter: ",", default: 0,
```
[the remark this answers is not in the document]
Comment (sbernhard) [reply lib/katello/repository_types/python.rb]: you are right. Thanks. Removed it. Copy & Paste issue.

**Review by ianballou — COMMENTED** [review]
Functionally this is working well when combined with https://github.com/Katello/katello/pull/11220

However, I found a UI bug. When you look at an existing  repository that has a "Keep latest packages" value, the UI shows it as blank. This issue seems related only to generic content  types that have integer fields.

I'm guessing the fix would live in /home/vagrant/katello/engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details-info.controller.js

**Review by ianballou — APPROVED** [review]
Works great, thanks!

File: lib/katello/repository_types/python.rb [hunk 2]
```diff
@@ -28,6 +28,9 @@
   generic_remote_option :package_types, title: N_("Package Types"), type: Array, input_type: "text", delimiter: ",", default: [],
                         description: N_("Package types to sync for Python content, separated by comma. Leave empty to get every package type. Package types are: bdist_dmg, bdist_dumb, bdist_egg, bdist_msi, bdist_rpm, bdist_wheel, bdist_wininst, sdist.")
 
+  generic_remote_option :keep_latest_packages, title: N_("Keep latest packages"), type: Integer, input_type: "number", default: 0,
```
Comment (ianballou) [inline lib/katello/repository_types/python.rb:31]:
```suggestion
  generic_remote_option :keep_latest_packages, title: N_("Keep latest packages"), type: :number, input_type: "number", default: 0,
```

**Review by ianballou — CHANGES_REQUESTED** [review]
Oh actually I missed one thing:

File: lib/katello/repository_types/python.rb (hunk 2 above)
Comment (sbernhard) [reply lib/katello/repository_types/python.rb]: Good point. Thank you!

**Review by ianballou — APPROVED** [review]
Thanks!

**Pull request merged**